### PR TITLE
chore!: support Laravel 11

### LINKED ISSUE #119: Laravel 11 support
Front logo [Front conversations](https://app.frontapp.com/open/top_15opi9)

### PR DESCRIPTION
**Pull request by stympy**
Add support for Laravel 11 and remove support for Laravel < 10.

Closes #119

### CODE REVIEW

**Comment by stympy** [comment]
@subzero10 It looks like we need to update the tests for the removal of `PHPUnit\Framework\MockObject\Builder\InvocationMocker::withConsecutive()` and `PHPUnit\Framework\MockObject\MockBuilder::setMethods()`

**Comment by stympy** [comment]
I have updated the tests and the dependencies. Along the way I also dropped support for Laravel < 10. 😁 I guess it's time for 4.0.0!

**Review by joshuap — APPROVED** [review]
This LGTM. I updated the PR title for conventional commits. I think that should cause a 4.0.0 release. I'd wait for a review from @subzero10 though. :)

File: .github/workflows/run-tests.yml
Comment (subzero10) [inline .github/workflows/run-tests.yml:28]: What was the reason for ditching Laravel v9?

Laravel v11 was just released a few days ago.
Laravel v9 just stopped receiving security fixes last month.

We are going to do a major version bump, so I think it's fine to say "we support versions that are supported by the community", but I'm wondering if there were breaking changes between the v9 and v10-11.

**Comment by subzero10** [comment]
> @subzero10 It looks like we need to update the tests for the removal of `PHPUnit\Framework\MockObject\Builder\InvocationMocker::withConsecutive()` and `PHPUnit\Framework\MockObject\MockBuilder::setMethods()`

Well, it seems you already handled that :)

**Comment by subzero10** [comment]
> This LGTM. I updated the PR title for conventional commits. I think that should cause a 4.0.0 release. I'd wait for a review from @subzero10 though. :)

We don't have automated releases for the PHP and Laravel packages. The [release process](https://github.com/honeybadger-io/honeybadger-laravel?tab=readme-ov-file#releasing) is simple enough that we didn't prioritize, but I guess we should work on that as well at some point.

**Review by subzero10 — APPROVED** [review]
I left a couple of non-blocking comments.

I tried to find a Laravel usage statistics report by version, but to my surprise I didn't find anything. I'm thinking that there would be very little adoption of v11 (it was just released this month), but more importantly I wanted to know the % distribution of v8 and v9. I guess we are safe since we are going with the major version bump.

**Comment by subzero10** [comment]
@stympy You can make the changes for the major version bump in this PR (modify `CHANGELOG.md` and `HoneybadgerLaravel.php`) or create a new PR just for the release after this one is merged.

File: .github/workflows/run-tests.yml
Comment (stympy) [reply .github/workflows/run-tests.yml:28]: I dropped 9 because of dependency problems between PHP 8.0, testbench, and phpunit, and I didn't want to spend the time to have conditional code in the tests to handle the older version of phpunit. Since Laravel 9 users can still use the 3.x version of the library, and since 9 is on the way out anyway, I think it's ok to drop it in a new major version of this library.

**Pull request merged**
